### ISSUE TITLE
Type error with strictTemplates: Type 'Subject<boolean>' is missing the following properties from type 'Reactive<boolean>'

### ISSUE THREAD

**Issue by vbraun** [dhilt/ngx-ui-scroll#242]
I'm getting the following type checker error with `@angular/core@11.0.8` and `ngx-ui-scroll@2.0.0-rc.1`
~~~
Error: projects/feature-lib/src/lib/foo.component.html:56:52 - error TS2322: Type '{ get: DatasourceGet; settings?: Settings | undefined; devSettings?: DevSettings | undefined; adapter: IAngularAdapter<unknown>; }' is not assignable to type 'IDatasource<unknown>'.
  The types of 'adapter.isLoading$' are incompatible between these types.
    Type 'Subject<boolean>' is missing the following properties from type 'Reactive<boolean>': value, id, options, subscriptions, and 5 more.

56                         *uiScroll="let obj of datasource"
                                                  ~~~~~~~~~~~~~

  projects/feature-lib/src/lib/foo.component.ts:59:18
    59     templateUrl: './foo.component.html',
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component FooComponent.
~~~
This only happens with strict template type checking enabled, that is, in tsconfig.json:
~~~
{
  "angularCompilerOptions": {
      "strictTemplates": true,
  },
[...]
~~~
If I disable strict type checking then no error is shown (perhaps unsurprisingly)

**Comment by dhilt** [comment]
@vbraun Thank you for trying v2 release candidate, this is very important. I was able to reproduce the issue with `strictTemplates` option enabled and pushed quick fix to [v2 PR](https://github.com/dhilt/ngx-ui-scroll/pull/238). It is available as **[ngx-ui-scroll@2.0.0-rc.3](https://www.npmjs.com/package/ngx-ui-scroll/v/2.0.0-rc.3)**. Please try and let me know if any problems!

Btw,  for better typing, if you are using Adapter API, you may specify the Adapter type as follows:

```ts
import { Datasource, IAdapter } from 'ngx-ui-scroll';

interface MyItem {
  id: number;
  myText: string;
  ...
}

...

  datasource = new Datasource<IAdapter<MyItem>>({ ... });
```

`MyItem` should reflect the structure of items that the Datasource deals with. It is "unknown" by default, and if not set, for example, `this.datasource.adapter.firstVisible.data.id` expression will produce typescript error: Object is of type 'unknown'. There are some Adapter props and methods dealing with `MyItem`, and if used, `MyItem` should be specified.

**Comment by vbraun** [comment]
Awesome! I can confirm that rc.3 fixes the issue for me.
